src/main/java/org/openjdk/jextract/impl/UnsupportedFilter.java

```diff
@@ -61,8 +59,8 @@ public UnsupportedFilter(PrintWriter errStream) {
         this.errStream = errStream;
     }
 
-    static String firstUnsupportedType(Type type) {
-        return type.accept(UNSUPPORTED_VISITOR, null);
+    static Type firstUnsupportedType(Type type, boolean allowVoid) {
```

Comment:
Note - the visitor now returns the unsupported type directly, instead of returning its string representation. This seems better.